src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs

```diff
@@ -920,17 +917,27 @@ public static void PropertyOrderIsRespected()
                             if (ti.Type == typeof(ClassWithExplicitOrderOfProperties))
                             {
                                 Assert.Equal(5, ti.Properties.Count);
+
                                 Assert.Equal("A", ti.Properties[0].Name);
                                 Assert.Equal("B", ti.Properties[1].Name);
                                 Assert.Equal("C", ti.Properties[2].Name);
```

Comment:
I don't think we should be pre-sorting anymore

Reply:
I was having similar thoughts, but ultimately decided against it for a few reasons:

1. The source generator already pre-sorts properties at compile-time.
2. The reflection resolver must precompute the property cache at construction time (it's needed when walking the type hierarchy). For performance reasons it only gets invalidated if the user explicitly requires the public property list. If the original property order were preserved it would necessitate duplicating some of the cache calculations at the `Configure` stage regardless of whether properties are being customized.
3. The order of reflection-based member enumeration can be nondeterministic depending on the runtime.

Reply:
1. that is a simple fix, not really an argument
2. it does not. It can move its logic to LateAdd and do exactly the same as source gen does
3. how do you know it's not deterministic? - regardless it's still not deterministic with pre-sorting by order

Reply:
> that is a simple fix, not really an argument

What? Why would we want to _regress_ source generator performance and force sorting at runtime?

> It can move its logic to LateAdd and do exactly the same as source gen does

Perhaps, that would require another round of refactoring though.

> how do you know it's not deterministic?

See https://github.com/dotnet/runtime/issues/2330#issuecomment-597326402, https://github.com/dotnet/runtime/issues/46272. It will be fixed in CLR for .NET 7 but since we're supporting other runtimes as well, for all intents and purposes the order of members is nondeterministic.

Reply:
> Why would we want to regress source generator performance and force sorting at runtime?

we can leave source gen as is but not pre-sort for reflection. source gen is separate resolver and can have different defaults than reflection

> for all intents and purposes the order of members is nondeterministic.

perhaps runtime is non-deterministic but we can be deterministic w.r.t to runtime and use runtime provided order